src/content/old-cleopatra-profile-format.js

```diff
@@ -310,6 +313,22 @@ function arrayFromArrayLikeObject<T>(obj: { [index: string]: T }): T[] {
   return result;
 }
 
+function _extractThreadList(profileJSON: OldCleopatraProfileJSON): OldCleopatraProfileThread[] {
+  if ('threads' in profileJSON) {
+    return arrayFromArrayLikeObject((profileJSON: any).threads);
+  }
+
+  // Ancient versions of the old cleopatra format did not have a threads list
+  // or markers. Instead, profileJSON was just the list of samples.
+  const oneThread = {
+    name: 'Main Thread',
+    markers: [],
+    samples: ((profileJSON: any): OldCleopatraSample[]),
+  };
+
+  return [oneThread];
```

Comment:
I managed to make flow happy without using `any` by reversing the initial condition:
```javascript
  if (Array.isArray(profileJSON)) {
    // Ancient versions of the old cleopatra format did not have a threads list
    // or markers. Instead, profileJSON was just the list of samples.
    const oneThread = {
      name: 'Main Thread',
      markers: [],
      samples: profileJSON,
    };

    return [oneThread];
  }

  return arrayFromArrayLikeObject(profileJSON.threads);
```